### ISSUE TITLE
[compile] refactor golden verification

### ISSUE THREAD

**Issue by pilkicTT** [tenstorrent/tt-forge-fe#146]
Currently, golden verification is broken, and remnants of previous golden implementation is sprinkled throughout the compile flow.

This refactor should centralize all the calculation/verification in one place and make it so that after each pass we can seamlessly call verification.

**Comment by nvukobratTT** [comment]
Yes. However, it's on our roadmap as the next item. @dgolubovicTT has already started an investigation into this one. 

Once we get more details, it'll potentially become a new milestone.

**Comment by skhatriTT** [comment]
@dgolubovicTT could you please provide a status on this?

**Comment by dgolubovicTT** [comment]
I enabled verification for both forward and backward pass in these PRs:
https://github.com/tenstorrent/tt-forge-fe/pull/1406
https://github.com/tenstorrent/tt-forge-fe/pull/1668

Currently, we have ability to target each compile phase and do verification. 
Therefore closing this one.
